### PR TITLE
DBZ-3238 Snapshot fails when reading TIME, DATE, DATETIME fields in mysql from ResultSet

### PR DESCRIPTION
**Pull request by jjiey**
[DBZ-3238](https://issues.redhat.com/browse/DBZ-3238)
Read the return value of mysql according to different protocols

### CODE REVIEW

**Comment by jjiey** [comment]
CC. @gunnarmorling @jpechane 
I have rebase my commit, but test still fails. What should I do?

**Comment by jjiey** [comment]
cc. @gunnarmorling @jpechane 
I have rebased my commit, and formatted the code, but test still fails.. What should I do?

**Comment by gunnarmorling** [comment]
There still seems to be a formatting issue; please run `mvn process-sources` to ensure that part is taken care of. Other than that, there are few intermittent test failures on CI unfortunately.

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/AbstractMysqlFieldReader.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package io.debezium.connector.mysql.legacy;
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/AbstractMysqlFieldReader.java:1]: License file header is missing.

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java [hunk 2]
```diff
@@ -809,6 +809,17 @@ public static GtidNewChannelPosition parse(String value, String defaultValue) {
                     + DatabaseHistory.CONFIGURATION_FIELD_PREFIX_STRING + "' string.")
             .withDefault(KafkaDatabaseHistory.class.getName());
 
+    public static final Field USE_CURSOR_FETCH = Field.create("useCursorFetch")
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java:812]: Do we really need that option? Can't we just rely on the pass-through `database.useCursorFetch` instead?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/AbstractMysqlFieldReader.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+
+/**
+ * Abstract class for decode MySQL return value according to different protocols.
+ *
+ * @author yangjie
+ */
+public abstract class AbstractMysqlFieldReader implements MysqlFieldReader {
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/AbstractMysqlFieldReader.java:17]: Hum, hum, that's interesting. Did you consciously decide to adjust the legacy implementation of the MySQL connector? The better place for all these changes would be `MySqlSnapshotChangeEventSource` and related code paths. I'm not opposing to _also_ adjusting the legacy implementation (though it is not going to be the focus of engineering going forward), but it should not be _only_ done there.

**Review by gunnarmorling — COMMENTED** [review]
Some first comments inline; mostly it's about making sure that the current implementation of the MySQL connector (`MySqlSnapshotChangeEventSource` etc.) gets adjusted. Also some tests should be added.

**Comment by jjiey** [comment]
@gunnarmorling Sorry, I'll fix it in `MySqlSnapshotChangeEventSource` immediately.

**Comment by jjiey** [comment]
Hi, @gunnarmorling I fixed this on both the legacy and current implementations, could you help to review this again? Thanks.

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/AbstractMysqlFieldReader.java (hunk 3 above)
Comment (jjiey) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/AbstractMysqlFieldReader.java]: I will fix this on current implementation.

**Comment by jpechane** [comment]
@jjiey Could you please reformat the code? Use `mvn process-sources`

**Comment by jjiey** [comment]
Hi, @jpechane Thanks for your suggestion.
Just finished, and I rebased all my commits. Finally the test succeeded.

**Comment by jpechane** [comment]
@jjiey Applied, thanks a lot for nice improvement!

**Pull request merged**